src/main/java/com/nashtech/exception/GlobalExceptionHandler.java

```
@@ -0,0 +1,57 @@
+package com.nashtech.exception;
+
+import com.azure.core.exception.ResourceNotFoundException;
+import com.azure.cosmos.CosmosException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+import java.time.LocalDateTime;
+
+/**
+ * A class that handles global exception handling for all
+ * REST controllers in the application.
+ */
+@RestControllerAdvice
+@Slf4j
+public class GlobalExceptionHandler {
+
+    /**
+     * Handles the DataNotFoundException globally.
+     * @param resourceNotFoundException the DataNotFoundException object
+     * @return a ResponseEntity object with an error message
+     * and HTTP status code
+     */
+    @ExceptionHandler(value = ResourceNotFoundException.class)
```

DataNotFound exception is not handled.